src/RestApi/StoreApi/Schemas/AbstractSchema.php

```diff
@@ -176,4 +176,20 @@ protected function prepare_money_response( $amount, $decimals = 2, $rounding_mod
 			)
 		);
 	}
+
+	/**
+	 * Prepares HTML based content, such as post titles and content, for the API response.
+	 *
+	 * The wptexturize, convert_chars, and trim functions are also used in the `the_title` filter.
+	 * The function wp_kses_post removes disallowed HTML tags.
+	 *
+	 * @param string|array $response Data to format.
+	 * @return string|array Formatted data.
+	 */
+	protected function prepare_html_response( $response ) {
+		if ( is_array( $response ) ) {
+			return array_map( [ $this, 'prepare_html_response' ], $response );
+		}
```

Comment:
Do you think there will ever be scenarios where `response` might be an object (or the array elements might be an object?) - should we either handle or guard against that if it is? I know we specify the expectation via the jsdocs, but without typehinting (which we can't do here) it's not enforced.

Reply:
I don't think there is currently, but we can guard by leaving objects alone.

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1597/commits/d50b2c24dca25918cc607744fea0b7a505a11fe0